File: BotsSDK/korebot/src/main/java/kore/botssdk/viewmodels/BaseViewModel.java
package kore.botssdk.viewmodels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.ViewModel;

import java.lang.ref.WeakReference;

import kore.botssdk.listener.BaseView;

public class BaseViewModel<N extends BaseView> extends ViewModel {
    private WeakReference<N> view = new WeakReference<>(null);

    protected void setView(N view) {
        this.view = new WeakReference<>(view);
    }

    protected N getView() {
        return view.get();
    }
}

File: BotSdk/botclient/src/main/java/com/kore/common/base/BaseViewModel.kt
package com.kore.common.base

import androidx.lifecycle.ViewModel
import kotlinx.coroutines.Job
import java.lang.ref.WeakReference


open class BaseViewModel<N : BaseView>() : ViewModel() {

    private var view: WeakReference<N?> = WeakReference(null)

    internal fun setView(view: N?) {
        this.view = WeakReference(view)
    }

    fun getView() = view.get()

    internal fun Job?.cancelIfActive() = this?.takeIf { it.isActive }?.cancel()
}